wagtailio/core/models/pages.py

```diff
@@ -130,3 +138,165 @@ class ContentPage(Page, HeroMixin, SocialMediaMixin, CrossPageMixin):
+        response = super().serve(request)
+        if self._is_htmx_request(request):
+            # We only return the fragment, rather than the whole page.
+            new_url = self.url + "?" + request.GET.urlencode()
+            response.headers["HX-Push"] = new_url
+        return response
+
+    def get_context(self, request):
+        context = super().get_context(request)
+        sectors = {}
+
+        for sector in self._filter_used_sectors():
+            if sector == request.GET.get("sector"):
+                sectors[sector] = "selected"
+            else:
+                sectors[sector] = ""
+
+        context |= {
+            "sectors": sectors,
+            "showcase_items": self._filtered_showcase_items(request.GET.get("sector")),
+            "current_sector": request.GET.get("sector") or "All",
+        }
+
+        return context
+
+
+class ShowcaseItem(Orderable):
+    title = models.CharField(max_length=255)
+    description = models.TextField(blank=True)
+    image = models.ForeignKey(
+        "images.WagtailIOImage",
+        models.SET_NULL,
+        null=True,
+        related_name="+",
+    )
+    alt_text = models.CharField(max_length=255, blank=True)
+    logo = models.ForeignKey(
+        "images.WagtailIOImage",
+        models.SET_NULL,
+        null=True,
+        blank=True,
+        related_name="+",
+    )
+    external_link = models.URLField(blank=True)
+    internal_link = models.ForeignKey(
+        "wagtailcore.Page",
+        models.SET_NULL,
+        null=True,
+        blank=True,
+        related_name="+",
+    )
+    link_text = models.CharField(max_length=255)
+
+    sector = models.ForeignKey(
+        "Sector",
+        models.CASCADE,
+        null=False,
+        blank=False,
+        related_name="+",
+    )
+
+    page = ParentalKey(
+        ShowcasePage,
+        related_name="showcase_items",
+        on_delete=models.CASCADE,
+    )
+
+    panels = [
+        FieldPanel("title"),
+        FieldPanel("sector"),
+        FieldPanel("description"),
+        FieldPanel("image"),
+        FieldPanel("alt_text"),
+        FieldPanel("logo"),
+        FieldPanel("external_link"),
+        FieldPanel("internal_link"),
+        FieldPanel("link_text"),
+    ]
+
+    def clean(self):
+        if not self.external_link and not self.internal_link:
+            raise ValidationError(
+                "You must specify either an external or internal link"
+            )
+        if self.external_link and self.internal_link:
+            raise ValidationError(
+                "You must specify either an external or internal link, not both"
+            )
+
+        if self.logo and not self.image:
+            raise ValidationError("You must specify an image when using a logo")
+
+    def __str__(self):
+        return self.title
+
+    @property
+    def url(self):
+        if self.external_link:
+            return self.external_link
+        return self.internal_link.url
```

Comment:
question: is there any chance this could be shifted to the template so as to make use of the request caching (ref: https://docs.wagtail.org/en/stable/advanced_topics/performance.html#page-urls)

Reply:
It can be, I guess this is just a preference of mine to keep the FE logic simple. Also, this site isn't a multi site so I think the query is the same regardless?

Reply:
I've made this change anyway, it could be someday.